crates/extensions/tedge-p11-server/src/pkcs11.rs

```diff
@@ -260,13 +246,41 @@ impl PkcsSigner {
 
         let key = session
             .find_objects(&key_template)
-            .map_err(|_| rustls::Error::General("pkcs11: Failed to find objects".to_owned()))?
+            .context("pkcs11: Failed to find objects")?
             .into_iter()
             .next()
-            .ok_or_else(|| rustls::Error::General("pkcs11: No signing key found".to_owned()))?;
-        debug!(?key, "selected key to sign");
+            .context("pkcs11: No signing key found")?;
+
+        // NOTE: cryptoki has AttributeType::AllowedMechanisms, but when i use it in get_attributes() with opensc-pkcs11
+        // module it gets ignored (not present or supported) and with softhsm2 module it panics(seems to be an issue
+        // with cryptoki, but regardless):
 
-        let mechanism = self.get_mechanism()?;
+        // thread 'main' panicked at library/core/src/panicking.rs:218:5:
+        // unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`
+        // note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+        // thread caused non-unwinding panic. aborting.
+        // Aborted (core dumped)
```

Comment:
This note is missing something. Would using `AttributeType::AllowedMechanisms` be better? Why? What are the impacts of using `AttributeType::KeyType`?

Oh, I see. The answer is a few lines below.